### PR TITLE
Improve readability

### PR DESCRIPTION
**Pull request by Wovchena**
1. stable_diffusion_1_5: return EXIT_FAILURE to stay consistent with the rest of main()
2. causal_lm 1. process()->select_next_tokens(). Although transformers name it process(), I believe select_next_tokens() is more verbose 2. return different values for different errors

### CODE REVIEW

File: text_generation/causal_lm/cpp/beam_search_causal_lm.cpp [hunk 1]
```diff
@@ -89,5 +89,5 @@ int main(int argc, char* argv[]) try {
     return 1;
 } catch (...) {
     std::cerr << "Non-exception object thrown\n";
-    return 1;
+    return 2;
```
Comment (ilya-lavrenov) [inline text_generation/causal_lm/cpp/beam_search_causal_lm.cpp:92]: should we be consistent and return either numbers or `EXIT_FAILURE` / `EXIT_SUCCESS` ?

File: .github/workflows/causal_lm_cpp.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ jobs:
       - name: Install OpenVINO
         run: |
           mkdir ./ov/
-          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/2023.3.0-13739-294cc6668c4/l_openvino_toolkit_ubuntu20_2023.3.0.dev20231219_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
+          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/2024.0.0-14049-0dc2dc5f602/l_openvino_toolkit_ubuntu20_2024.0.0.dev20240118_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:36]: is it supposed to return back to 2023.3 once official release is out?

Comment (Wovchena) [reply .github/workflows/causal_lm_cpp.yml:36]: I don't have objections to set 23.3 back. The problem is that there's no longer nightly 23.3

Comment (ilya-lavrenov) [reply .github/workflows/causal_lm_cpp.yml:36]: - we are on releases/2023/3 branch, so it's expected to test against corresponding release
- current nightly release will be removed in future as well and you will have to update it one more time.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**